bika/lims/exportimport/instruments/varian/vistapro/icp.py

```diff
@@ -0,0 +1,265 @@
+        result = str(result)
+        if result.startswith('--') or result == '' or result == 'ND':
+            return 0.0
+
+        try:
+            result = float(result)
+            if result < 0.0:
+                result = 0.0
+        except ValueError:
+            self.err(
+                "No valid number ${result} in column (${column_name})",
+                mapping={"result": result,
+                         "column_name": column_name},
+                numline=self._numline, line=line)
+            return
+        return result
+
+
+class VistaPROICPImporter(AnalysisResultsImporter):
+    """ Importer
+    """
+
+    def __init__(self, parser, context, idsearchcriteria, override,
+                 allowed_ar_states=None, allowed_analysis_states=None,
+                 instrument_uid=None):
+
+        AnalysisResultsImporter.__init__(self,
+                                         parser,
+                                         context,
+                                         idsearchcriteria,
+                                         override,
+                                         allowed_ar_states,
+                                         allowed_analysis_states,
+                                         instrument_uid)
+
+
+def Import(context, request):
+    """ Import Form
+    """
+    form = request.form
+    # form['file'] sometimes returns a list
+    infile = form['instrument_results_file'][0] if \
+        isinstance(form['instrument_results_file'], list) \
+        else form['instrument_results_file']
+    override = form['results_override']
+    artoapply = form['artoapply']
+    sample = form.get('sample', 'requestid')
+    instrument = form.get('instrument', None)
+    errors = []
+    logs = []
+    warns = []
+
+    # Load the most suitable parser according to file extension/options/etc...
+    parser = None
+    if not hasattr(infile, 'filename'):
+        errors.append(_("No file selected"))
+
+    parser = VistaPROICPParser(infile)
+    if parser:
+        # Load the importer
+        status = ['sample_received', 'attachment_due', 'to_be_verified']
+        if artoapply == 'received':
+            status = ['sample_received']
+        elif artoapply == 'received_tobeverified':
+            status = ['sample_received', 'attachment_due', 'to_be_verified']
+
+        over = [False, False]
+        if override == 'nooverride':
+            over = [False, False]
+        elif override == 'override':
+            over = [True, False]
+        elif override == 'overrideempty':
+            over = [True, True]
+
+        sam = ['getRequestID', 'getSampleID', 'getClientSampleID']
+        if sample == 'requestid':
+            sam = ['getRequestID']
+        if sample == 'sampleid':
+            sam = ['getSampleID']
+        elif sample == 'clientsid':
+            sam = ['getClientSampleID']
+        elif sample == 'sample_clientsid':
+            sam = ['getSampleID', 'getClientSampleID']
```

Comment:
I've created a file called instruments/utils.py because I could not import get_instrument_import_search_criteria when it is on the instruments.__init__.py since icp.py is also imported on instruments.__init__.py